spec/filters/csv_spec.rb

```diff
@@ -1,175 +1,160 @@
 # encoding: utf-8
-
 require "logstash/devutils/rspec/spec_helper"
 require "logstash/filters/csv"
 
 describe LogStash::Filters::CSV do
 
+  subject(:plugin) { LogStash::Filters::CSV.new(config) }
+  let(:config) { Hash.new }
+
+  let(:doc) { ""}
+  let(:event) { LogStash::Event.new("message" => doc) }
+
+
+  before(:each) do
+    plugin.register
+  end
+
   describe "all defaults" do
-    # The logstash config goes here.
-    # At this time, only filters are supported.
-    config <<-CONFIG
-      filter {
-        csv { }
-      }
-    CONFIG
-
-    sample "big,bird,sesame street" do
-      insist { subject["column1"] } == "big"
-      insist { subject["column2"] } == "bird"
-      insist { subject["column3"] } == "sesame street"
+
+    let(:doc) { "big,bird,sesame street" }
+
+    it "extract all the values" do
+      plugin.filter(event)
+      expect(event["column1"]).to eq("big")
+      expect(event["column2"]).to eq("bird")
+      expect(event["column3"]).to eq("sesame street")
     end
   end
 
   describe "custom separator" do
-    config <<-CONFIG
-      filter {
-        csv {
-          separator => ";"
-        }
-      }
-    CONFIG
-
-    sample "big,bird;sesame street" do
-      insist { subject["column1"] } == "big,bird"
-      insist { subject["column2"] } == "sesame street"
+    let(:doc) { "big,bird;sesame street" }
+
+    let(:config) do
+      { "separator" => ";" }
+    end
```

Comment:
Consider updating the config hash you created in line 8 e.g.
In the `it block` add `config.update("separator" => ";")`

Only if you want to. :smile: